### ISSUE TITLE
Build of api failed: missing api-params.json

### ISSUE THREAD

**Issue by OlgaMaslova** [aws-amplify/amplify-cli#9167]
**Which Category is your question related to?**
Build amplify with the newest version of amplify-cli

**Amplify CLI Version** 
7.5.5

**What AWS Services are you utilizing?**
amplify

App ID and region:
do5lfh5kuwuas, eu-west-1

**Provide additional details e.g. code snippets**
`
✔ Initialized provider successfully.
2021-12-01T14:48:28.059Z [INFO]: 🛑 Error reading api-params.json file for MYAPI resource
2021-12-01T14:48:28.118Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2021-12-01T14:48:28.120Z [WARNING]: ✖ There was an error initializing your environment.
2021-12-01T14:48:28.123Z [INFO]: [0mError: File at path: '/codebuild/output/src918130845/src/***/amplify/backend/api/MYAPI/api-params.json' does not exist[0m
                                 [0m    at Function.JSONUtilities.readJson (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/jsonUtilities.ts:25:15)[0m
                                 [0m    at ApigwInputState.migrateApigwResource (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/provider-utils/awscloudformation/apigw-input-state.ts:127:44)[0m
                                 [0m    at runMicrotasks (<anonymous>)[0m
                                 [0m    at processTicksAndRejections (internal/process/task_queues.js:97:5)[0m
2021-12-01T14:48:28.147Z [ERROR]: !!! Build failed
2021-12-01T14:48:28.147Z [ERROR]: !!! Non-Zero Exit Code detected
`

I am having this issue with the latest version of amplify-cli. No problem with 7.5.4.

Indeed the required file does not exist and never existed! No problem until today.

**Comment by Straubulous** [comment]
OlgaMaslova & jeffbking - can try force push with 7.5.5?

**Comment by jhockett** [comment]
@OlgaMaslova do you know what version of Amplify CLI you created `MYAPI` with?

**Comment by OlgaMaslova** [comment]
@jhockett it has been more than one and a half year ago... So, not really.

**Comment by giorgiocarniel** [comment]
Same problem here, everything worked until two days ago.

First error noticed this morning was
`File at path: '/codebuild/output/src781572205/src/preorder-aws/amplify/backend/api/AdminQueries/api-params.json' does not exist`

Then when retrying the error became this (without touching the codebase)
```
TypeError: Cannot read property 'updateamplifyMetaAfterResourceUpdate' of undefined[0m
[0m    at ApigwInputState.updateAdminQueriesResource (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/provider-utils/awscloudformation/apigw-input-state.ts:69:32)[0m
 [0m    at migrateResourceNonContainer (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/provider-utils/awscloudformation/index.ts:295:10)[0m
 [0m    at async Promise.all (index 0)[0m
 [0m    at migrate (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/index.ts:77:3)[0m
 [0m    at loadApiCliInputs (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts:260:7)[0m
 [0m    at consolidateApiGatewayPolicies (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts:193:23)[0m
 [0m    at createEnvLevelConstructs (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/utils/env-level-constructs.ts:24:5)[0m
 [0m    at Object.run (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/push-resources.ts:114:5)
```

Noticed the error just on CI/CD, I've not tried it locally yet.
My API was created more than 1 year ago.

**Comment by jhockett** [comment]
@giorgiocarniel that issue is slightly different. https://github.com/aws-amplify/amplify-cli/pull/9182 should fix your issue.
 
 @OlgaMaslova can you send your amplify folder to amplify-cli@amazon.com so we can better debug? Please include the issue number in the subject and redact any sensitive information.

**Comment by OlgaMaslova** [comment]
@jhockett  the mail has been sent.

**Comment by jhockett** [comment]
@OlgaMaslova yeah that's most likely enough. Could you also include the `amplify-meta.json` file?

**Comment by jhockett** [comment]
@OlgaMaslova got your email, thanks for sending it. I'm not sure why you're seeing an error from `MYAPI`, as it doesn't exist in `amplify-meta.json` from what I can see.

**Comment by levinskipolish** [comment]
@giorgiocarniel the very same thing happened to me!

I tried:
```
1 - re-deploy the app from a previous successful deployment.
2 - pull down the backend to refresh any uncaught dependencies ––Sometime it helps.
```
But I still have the very same issue!
```
2021-12-02T20:40:01.362Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2021-12-02T20:40:01.372Z [WARNING]: ✖ There was an error initializing your environment.
2021-12-02T20:40:01.385Z [INFO]: [0mTypeError: Cannot read property 'updateamplifyMetaAfterResourceUpdate' of undefined[0m
[0m    at ApigwInputState.updateAdminQueriesResource (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/provider-utils/awscloudformation/apigw-input-state.ts:69:32)[0m
[0m    at migrateResourceNonContainer (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/provider-utils/awscloudformation/index.ts:295:10)[0m
[0m    at async Promise.all (index 0)[0m
[0m    at migrate (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-api/src/index.ts:77:3)[0m
[0m    at loadApiCliInputs (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts:260:7)[0m
 [0m    at consolidateApiGatewayPolicies (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts:193:23)[0m
 [0m    at createEnvLevelConstructs (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/utils/env-level-constructs.ts:24:5)[0m
 [0m    at Object.run (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/push-resources.ts:114:5)[0m
```

**Comment by OlgaMaslova** [comment]
@jhockett the api's real name is AdminQueries, that the folder I sent

**Comment by jhockett** [comment]
@levinskipolish if you push a second time with the CLI, the issue should resolve itself. That error is little different from what @OlgaMaslova is running into.

**Comment by jhockett** [comment]
@OlgaMaslova was the parent folder renamed? `AdminQueries` is a special name for the CLI, so it shouldn't be changed.

**Comment by OlgaMaslova** [comment]
@jhockett that's the structure that has been there for more than a year...
![image](https://user-images.githubusercontent.com/29795366/144502592-14eda4b3-040b-410e-8d70-934306851f5b.png)

**Comment by jhockett** [comment]
@OlgaMaslova I see, thanks for the clarification. We have merged a fix and this will be fixed in the next release.

**Comment by levinskipolish** [comment]
> @levinskipolish if you push a second time with the CLI, the issue should resolve itself. That error is little different from what @OlgaMaslova is running into.

Still getting the very same issue! :/

**Comment by jhockett** [comment]
@levinskipolish what version of the CLI are you using?

**Comment by levinskipolish** [comment]
@jeffbking my cli version is `7.6.1`
I also see this "Patching NPM package..." during the deployment, under the Cloning Repository phase.

```
2021-12-02T22:09:58.959Z [INFO]: # Patching NPM package '@aws-amplify/cli' from 4.29.4 to 7.6.1...
2021-12-02T22:10:37.659Z [INFO]: npm WARN deprecated @graphql-toolkit/common@0.9.7: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2021-12-02T22:10:43.770Z [INFO]: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
2021-12-02T22:10:43.796Z [INFO]: npm WARN
2021-12-02T22:10:43.796Z [INFO]: deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
2021-12-02T22:10:44.687Z [INFO]: npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
2021-12-02T22:10:46.923Z [INFO]: npm
2021-12-02T22:10:46.924Z [INFO]: WARN deprecated @graphql-toolkit/common@0.6.6: GraphQL Toolkit is deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-toolkit for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2021-12-02T22:10:47.573Z [INFO]: npm WARN deprecated graphql-import@0.7.1: GraphQL Import has been deprecated and merged into GraphQL Tools, so it will no longer get updates. Use GraphQL Tools instead to stay up-to-date! Check out https://www.graphql-tools.com/docs/migration-from-import for migration and https://the-guild.dev/blog/graphql-tools-v6 for new changes.
2021-12-02T22:10:56.542Z [INFO]: npm WARN
2021-12-02T22:10:56.546Z [INFO]: deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
2021-12-02T22:11:05.941Z [INFO]: npm WARN deprecated event-to-promise@0.8.0: Use promise-toolbox/fromEvent instead
2021-12-02T22:12:13.027Z [INFO]: npm
2021-12-02T22:12:13.028Z [INFO]: WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^2.1.2 (node_modules/@aws-amplify/cli/node_modules/jest-haste-map/node_modules/fsevents):
                                 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
2021-12-02T22:12:13.114Z [INFO]: # Done patching NPM package '@aws-amplify/cli' to version 7.6.1
```

**Comment by jhockett** [comment]
@levinskipolish I see, I didn't realize you were facing this issue on the console. The upcoming release should resolve this.

**Comment by levinskipolish** [comment]
@jhockett Awesome! Looking forward to see the release notes. I'm curious to know how this issue came about. I'm sure it affects a good amount of application out there. Where can I learn more about the upcoming releases?

**Comment by jhockett** [comment]
@levinskipolish release notes are published here: https://github.com/aws-amplify/amplify-cli/releases

Here's the PR that fixes the issue: https://github.com/aws-amplify/amplify-cli/pull/9182

**Comment by jhockett** [comment]
A fix for this was released in version 7.6.3.
